dependencies/O2PhysicsCompileFlags.cmake

```diff
@@ -11,13 +11,53 @@
+
+# Enabled warnings supported by GCC only
+set(O2PHYSICS_WARNINGS_GCC "aggressive-loop-optimizations;analyzer-double-fclose;analyzer-double-free;analyzer-exposure-through-output-file;analyzer-file-leak;analyzer-free-of-non-heap;analyzer-malloc-leak;analyzer-mismatching-deallocation;analyzer-null-argument;analyzer-null-dereference;analyzer-possible-null-argument;analyzer-possible-null-dereference;analyzer-shift-count-negative;analyzer-shift-count-overflow;analyzer-stale-setjmp-buffer;analyzer-tainted-allocation-size;analyzer-tainted-array-index;analyzer-tainted-divisor;analyzer-tainted-offset;analyzer-tainted-size;analyzer-unsafe-call-within-signal-handler;analyzer-use-after-free;analyzer-use-of-pointer-in-stale-stack-frame;analyzer-use-of-uninitialized-value;analyzer-write-to-const;analyzer-write-to-string-literal;array-bounds=1;array-compare;array-parameter=2;bool-compare;builtin-declaration-mismatch;cannot-profile;cast-function-type;catch-value=1;class-memaccess;clobbered;coverage-invalid-line-number;coverage-mismatch;dangling-pointer=2;format-contains-nul;format-diag;format-overflow=1;format-truncation=1;format=1;if-not-aligned;implicit-fallthrough=3;inherited-variadic-ctor;init-list-lifetime;init-self;interference-size;invalid-memory-model;literal-suffix;lto-type-mismatch;maybe-uninitialized;memset-elt-size;mismatched-dealloc;missing-attributes;missing-profile;missing-requires;missing-template-keyword;multistatement-macros;non-template-friend;nonnull-compare;openmp-simd;overflow;packed-bitfield-compat;packed-not-aligned;pmf-conversions;prio-ctor-dtor;restrict;return-local-addr;scalar-storage-order;sized-deallocation;strict-aliasing=3;strict-overflow=1;stringop-overflow=2;stringop-overread;stringop-truncation;subobject-linkage;switch-outside-range;switch-unreachable;sync-nand;terminate;tsan;type-limits;unused-local-typedefs;use-after-free=2;virtual-move-assign;vla-parameter;volatile-register-var;zero-length-bounds")
```

Comment:
I am not sure this is a great idea, actually. Can't you simply enable larger classes of warning?

Reply:
As listed in the PR description, these should correspond to the union of default, `-Wall` and `-Wextra` for the respective compilers. If we enable larger classes, don't we risk getting new errors from new warnings added into those classes?

Reply:
Following up on our private discussion, I have simplified the flags by treating all enabled warnings as errors and defining lists for the excluded ones. See 75fb9c0.